blackboxopt/optimizers/staged/optimizer.py

```diff
@@ -46,20 +46,27 @@ def __init__(
         self.evaluation_uuid_to_iteration: Dict[str, int] = {}
         self.pending_configurations: Dict[str, EvaluationSpecification] = {}
 
-    def report_evaluation(self, evaluation: Evaluation) -> None:
-        super().report_evaluation(evaluation)
+    def report_evaluations(self, evaluations: Iterable[Evaluation]) -> None:
+        super().report_evaluations(evaluations)
 
-        evaluation_specification_id = evaluation.optimizer_info.get("id")
-        if evaluation_specification_id is None:
+        missing_ids_count = sum(e.optimizer_info.get("id") is None for e in evaluations)
+        if missing_ids_count > 0:
             raise ValueError(
-                "Missing evaluation specification ID in optimizer info. Did you try to "
-                + "report an evaluation for a configuration which the optimizer did not"
-                " pick? This is not supported at the moment."
+                f"{missing_ids_count} evaluation(s) miss an evaluation specification ID"
+                + "in the optimizer info. "
+                + "Did you try to report evaluations for configurations which the "
+                + "optimizer did no pick? This is not supported at the moment."
```

Comment:
I'm not sure, what the best behavior here. I considered three options:
- process all evaluations that are valid, skip only the invalid ones (ignore).
- process all evaluations until an invalid one, and ignore the following ones (abort).
- process none evaluations, even if only a single one is invalid (strict).

Here I proposed the "strict" version, because I think it gives the most control to user. Also it might be more difficult to _identify_ the invalid evaluations if some are processed and others are not. 

What do you think, @sfalkner, @LGro ?